### PR TITLE
MySQL e2e-tests

### PR DESCRIPTION
**Pull request by poojantcs**

### CODE REVIEW

File: mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License..
+
+Feature: MySQL Source - Design time scenarios
+
+  @TS-MYSQL-SOURCE-DSGN-01 @MYSQL_SOURCE_TEST
+  Scenario: Verify user can validate successfully when plugin is configured for table with connection and basic details
```
Comment (Vipinofficial11) [inline mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature]: We can rephrase the above scenario to be more understandable/readable -  Verify user is able to validate the plugin when configured for basic Section.

File: mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    Then Replace input plugin property: "password" with value: "password" for Credentials and Authorization related fields
+    Then Enter input plugin property: "referenceName" with value: "sourceRef"
+    Then Replace input plugin property: "database" with value: "databaseName"
+    Then Enter textarea plugin property: "importQuery" with value: "selectQuery"
+    Then Click on the Get Schema button
+    Then Verify the Output Schema matches the Expected Schema: "outputSchema"
+    Then Validate "MySQL" plugin properties
+
+  @TS-MYSQL-SOURCE-DSGN-02 @MYSQL_SOURCE_TEST @CONNECTION
+  Scenario: Verify the MySQL Source plugin using connection manager functionality
```
Comment (Vipinofficial11) [inline mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature]: Rephrase - Verify user is able validate the plugin with connection management functionality.

File: mysql-plugin/src/e2e-test/features/mysqlsource/DesignTimeValidation.feature [hunk 3]
```diff
@@ -0,0 +1,149 @@
+# Copyright © 2022 Cask Data, Inc.
```
Comment (Vipinofficial11) [inline mysql-plugin/src/e2e-test/features/mysqlsource/DesignTimeValidation.feature]: Make it 2023 in all files

File: mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    Then Replace input plugin property: "password" with value: "password" for Credentials and Authorization related fields
+    Then Enter input plugin property: "referenceName" with value: "sourceRef"
+    Then Replace input plugin property: "database" with value: "databaseName"
+    Then Enter textarea plugin property: "importQuery" with value: "selectQuery"
+    Then Click plugin property: "useCompression"
+    Then Click plugin property: "useAnsiQuotes"
+    Then Click plugin property: "autoReconnect"
+    Then Click on the Get Schema button
+    Then Verify the Output Schema matches the Expected Schema: "outputSchema"
+    Then Validate "MySQL" plugin properties
```
Comment (Vipinofficial11) [inline mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature]: Add a new line at the end of this file

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/locators/MySQLPropertiesPage.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+import io.cdap.e2e.utils.SeleniumDriver;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+
+public class MySQLPropertiesPage {
+    @FindBy(how = How.XPATH, using = "//div[@data-cy='connector-MySQL']")
+    public static WebElement connectorMysql;
```
Comment (Vipinofficial11) [inline mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/locators/MySQLPropertiesPage.java:32]: 2 space indentation(spacing) is being followed in this project. please fix the indentation in this class.

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/actions/MySQLPropertiesPageActions.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    public static void clickOnMysqlConnectionButton() {
+        ElementHelper.clickOnElement(MySQLPropertiesPage.connectorMysql);
+    }
+
+    public static void clickOnMySQLConnection() {
+        String connectionName = PluginPropertyUtils.pluginProp("connection.name");
+        ElementHelper.clickOnElement(MySQLPropertiesPage.MySQLConnection(connectionName));
+    }
+}
+
```
Comment (Vipinofficial11) [inline mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/actions/MySQLPropertiesPageActions.java]: Use 2 space indentation in this class as well. Also add package-info file for both Action and Locator class.

File: mysql-plugin/src/e2e-test/resources/pluginParameters.properties [hunk 7]
```diff
@@ -1,12 +1,36 @@
...
 username=MYSQL_USERNAME
+invalid.username=root1
 password=MYSQL_PASSWORD
-databaseName=sakila
+databaseName=testDatabase
 port=MYSQL_PORT
+numberofsplits=abc
+numberofsplitsgenerate=
+numbersplitsgenerate=2
+invalid.tablename=123#
```
Comment (Vipinofficial11) [inline mysql-plugin/src/e2e-test/resources/pluginParameters.properties:13]: Separate the properties using comments. For example 
#AdvancedProperties
numberofsplits=abc

#Invalid Properties
property1=
property2=

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License..
+
+Feature: MySQL Sink - Design time scenarios
+
+  @TS-MYSQL-SINK-DSGN-01 @MYSQL_SOURCE_TEST
+  Scenario: Verify user can validate successfully when sink plugin is configured for table with basic details
```
Comment (Vipinofficial11) [inline mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature:19]: Have a look at the Scenarios in all the feature files, Just make them more understandable and readable.

File: mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature (hunk 1 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature]: done

File: mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature (hunk 2 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature]: done

File: mysql-plugin/src/e2e-test/features/mysqlsource/DesignTimeValidation.feature (hunk 3 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsource/DesignTimeValidation.feature]: done

File: mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature (hunk 4 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsource/DesignTime.feature]: done

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/locators/MySQLPropertiesPage.java (hunk 5 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/locators/MySQLPropertiesPage.java:32]: DONE

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/actions/MySQLPropertiesPageActions.java (hunk 6 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/actions/MySQLPropertiesPageActions.java]: DONE

File: mysql-plugin/src/e2e-test/resources/pluginParameters.properties (hunk 7 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/resources/pluginParameters.properties:13]: done

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature (hunk 8 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature:19]: done

**Comment by Vipinofficial11** [comment]
Closing the PR as the external review is done and its already Approved.

**Pull request closed without merging**